lib/integrations/awsoidc/aws_app_access_iam_config.go

```diff
@@ -0,0 +1,124 @@
+	//Region string
+
+	// IntegrationRole is the Integration's AWS Role used to set up Teleport as an OIDC IdP.
+	IntegrationRole string
+
+	// IntegrationRoleAWSAppAccessPolicy is the Policy Name that is created to allow access to call AWS APIs.
+	// Defaults to AWSAppAccess
+	IntegrationRoleAWSAppAccessPolicy string
+}
+
+// CheckAndSetDefaults ensures the required fields are present.
+func (r *AWSAppAccessConfigureRequest) CheckAndSetDefaults() error {
+	if r.IntegrationRole == "" {
+		return trace.BadParameter("integration role is required")
+	}
+
+	if r.IntegrationRoleAWSAppAccessPolicy == "" {
+		r.IntegrationRoleAWSAppAccessPolicy = defaultPolicyNameForAWSAppAccess
+	}
+
+	return nil
+}
+
+// AWSAppAccessConfigureClient describes the required methods to create the IAM Policies required for AWS App Access.
+type AWSAppAccessConfigureClient interface {
+	// PutRolePolicy creates or replaces a Policy by its name in a IAM Role.
+	PutRolePolicy(ctx context.Context, params *iam.PutRolePolicyInput, optFns ...func(*iam.Options)) (*iam.PutRolePolicyOutput, error)
+}
+
+type defaultAWSAppAccessConfigureClient struct {
+	*iam.Client
+}
+
+// NewAWSAppAccessConfigureClient creates a new AWSAppAccessConfigureClient.
+func NewAWSAppAccessConfigureClient(ctx context.Context) (AWSAppAccessConfigureClient, error) {
+	cfg, err := config.LoadDefaultConfig(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	if cfg.Region == "" {
+		// IAM Service does not support regions, however a value is required:
+		// https://github.com/aws/aws-sdk-go-v2/issues/1778#issuecomment-1210031692
+		// Providing an invalid region here, ensures the service uses the default AWS Partition.
+		cfg.Region = " "
+	}
+
+	return &defaultAWSAppAccessConfigureClient{
+		Client: iam.NewFromConfig(cfg),
+	}, nil
+}
+
+// ConfigureAWSAppAccess set ups the roles required for AWS App Access.
+// It creates an embedded policy with the following permissions:
+// - sts:AssumeRole
+//
+// The following actions must be allowed by the IAM Role assigned in the Client.
+//   - iam:PutRolePolicy
+func ConfigureAWSAppAccess(ctx context.Context, clt AWSAppAccessConfigureClient, req AWSAppAccessConfigureRequest) error {
+	if err := req.CheckAndSetDefaults(); err != nil {
+		return trace.Wrap(err)
+	}
+
+	awsAppAccessPolicyDocument, err := awslib.NewPolicyDocument(
+		awslib.StatementForAWSAppAccess(),
+	).Marshal()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	_, err = clt.PutRolePolicy(ctx, &iam.PutRolePolicyInput{
+		PolicyName:     &req.IntegrationRoleAWSAppAccessPolicy,
+		RoleName:       &req.IntegrationRole,
+		PolicyDocument: &awsAppAccessPolicyDocument,
+	})
+	if err != nil {
+		if trace.IsNotFound(awslib.ConvertIAMv2Error(err)) {
+			return trace.NotFound("role %q not found.", req.IntegrationRole)
+		}
+		return trace.Wrap(err)
+	}
+
+	slog.With("policy", req.IntegrationRoleAWSAppAccessPolicy).With("role", req.IntegrationRole).InfoContext(ctx, "IAM Inline Policy added to IAM Role.")
```

Comment:
"With" creates a new logger, that is not necessary here:

also nit: Don't end single-sentence logs with punctuation: https://github.com/gravitational/teleport/blob/master/rfd/0154-logging-guidelines.md#logging-standards

```suggestion
	slog.InfoContext(ctx,
		"IAM Inline Policy added to IAM Role",
		"policy", req.IntegrationRoleAWSAppAccessPolicy,
		"role", req.IntegrationRole,
	)
```